crates/grit-pattern-matcher/README.md

```diff
@@ -0,0 +1,9 @@
+# `grit-pattern-matcher`
+
+This crate contains all the pattern definitions that are at the heart of the
+GritQL engine. There's a `Matcher` trait that's implemented by the patterns,
+which implements the matching logic.
+
+It is important this crate stays free of TreeSitter dependencies, since it is
```

Comment:
It would be a good opportunity to add a Grit rule to enforce that actually, in `.grit`, otherwise I'm sure someone will accidentally try. 

Even something basic like `tree_sitter` should be sufficient.

Reply:
Done. Do I need to add anything to make sure CI runs it, or does it always check all patterns in `grit.yaml`?

Reply:
That's all you need to do, grit check picks it up automatically.